### DISCUSSION TITLE
Enter page and then full screening browser?

### DISCUSSION THREAD

**Discussion by ThonAfterPy** [zauberzeug/nicegui#627] [Q&A] [answered]
Hello. Basically, what I am trying to do is put the browser in full screen after opening a ui.page using this code:
```py
@ui.page('/full_screen')
async def alexa_mode(client: Client):
    await client.connected(5)
    await ui.run_javascript('if (!document.fullscreenElement) {document.documentElement.requestFullscreen();} else if (document.exitFullscreen) {document.exitFullscreen();}', respond=False)
    
ui.button("Go to fullscreen page", on_click=lambda: ui.open("/alexa_mode"))
```

The problem is that when running the code, I get an "TypeError: Permissions check failed" in console, which I believe is because you are not allowed to automatically full screen the webpage.

I'm wondering if there is anything I can do to get around this, or another solution that still does the same thing. One thing that I tried was adding a button to the new page which, upon clicked, would call a function put the browser in fullscreen. This worked, but was not very nice to have to click a button just to put the browser in full screen.

**Comment by rodja** [comment, marked as answer]
Interesting question and without a simple solution. The Fullscreen API requires a user gesture to activate, like a click or keypress, to prevent unwanted fullscreen behavior.

**Comment by rodja** [comment]
You can now use [the native mode](https://nicegui.io/documentation#native_mode) to show your app in fullscreen.
